onnxruntime/test/framework/float_4_test.cc

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !defined(DISABLE_FLOAT4_TYPES)
+
+#include <vector>
+#include <map>
+
+#ifdef USE_CUDA
+// Needed for CUDA_VERSION check in float4.h
+#include <cuda.h>
```

Comment:
Including this on CUDA enabled builds will ensure that CUDA based converion routines are used in CUDA12.8+ builds as CUDA_VERSION (from cuda.h) is used as a gating check in float4.h